libsolidity/analysis/DeclarationContainer.cpp

```diff
@@ -115,3 +115,57 @@ std::vector<Declaration const*> DeclarationContainer::resolveName(ASTString cons
 		return m_enclosingContainer->resolveName(_name, true);
 	return vector<Declaration const*>({});
 }
+
+std::vector<ASTString> DeclarationContainer::similarNames(ASTString const& _name) const
+{
+	std::vector<ASTString> similar;
+
+	for (const auto & declaration: m_declarations)
+	{
+		const string& declarationName = declaration.first;
+		if (DeclarationContainer::areSimilarNames(_name, declarationName))
+			similar.push_back(declarationName);
+	}
+
+	if (m_enclosingContainer)
+	{
+		std::vector<ASTString> enclosingSimilar = m_enclosingContainer->similarNames(_name);
+		similar.insert(similar.end(), enclosingSimilar.begin(), enclosingSimilar.end());
```

Comment:
I wonder which is faster. The current way or doing the parent first and the `push_front` to extend that vector?

Reply:
I think this is reasonably fast. push_front on vector is quite slow (though we can use list/deque to get around that). Inserting at end like here is [efficient](https://stackoverflow.com/questions/2208293/what-is-the-most-efficient-way-to-append-one-stdvector-to-the-end-of-another). 

It can be made slightly faster by avoiding creation of multiple vectors. We can create a single vector and pass it by reference as a parameter to the function. Or we can use a list and use splice to append at end. But I am not sure it will result in any substantial gain.

PS: Thanks for the tips :) will remove std:: and add change log to all my PRs.